src/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php

```diff
@@ -0,0 +1,174 @@
+	 *
+	 * @since 6.5.0
+	 *
+	 * @access private
+	 *
+	 * @param string $new_content The string to replace the content between the matching tags.
+	 * @return bool Whether the content was successfully replaced.
+	 */
+	public function set_content_between_balanced_tags( string $new_content ): bool {
+		$this->get_updated_html();
+
+		$bookmarks = $this->get_balanced_tag_bookmarks();
+		if ( ! $bookmarks ) {
+			return false;
+		}
+		list( $start_name, $end_name ) = $bookmarks;
+
+		$start = $this->bookmarks[ $start_name ]->start + $this->bookmarks[ $start_name ]->length + 1;
+		$end   = $this->bookmarks[ $end_name ]->start;
+
+		$this->seek( $start_name );
+		$this->release_bookmark( $start_name );
+		$this->release_bookmark( $end_name );
+
+		$this->lexical_updates[] = new WP_HTML_Text_Replacement( $start, $end - $start, esc_html( $new_content ) );
+		return true;
+	}
+
+	/**
+	 * Returns a pair of bookmarks for the current opening tag and the matching
+	 * closing tag.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array|null A pair of bookmarks, or null if there's no matching closing tag.
+	 */
+	private function get_balanced_tag_bookmarks() {
+		static $i   = 0;
+		$start_name = 'start_of_balanced_tag_' . ++$i;
+
+		$this->set_bookmark( $start_name );
+		if ( ! $this->next_balanced_closer() ) {
+			$this->release_bookmark( $start_name );
+			return null;
+		}
+
+		$end_name = 'end_of_balanced_tag_' . ++$i;
+		$this->set_bookmark( $end_name );
+
+		return array( $start_name, $end_name );
+	}
+
+	/**
+	 * Finds the matching closing tag for an opening tag.
+	 *
+	 * When called while the processor is on an open tag, it traverses the HTML
+	 * until it finds the matching closer tag, respecting any in-between content,
+	 * including nested tags of the same name. Returns false when called on a
+	 * closer tag, a tag that doesn't have a closer tag (void), a tag that
+	 * doesn't visit the closer tag, or if no matching closing tag was found.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return bool Whether a matching closing tag was found.
+	 */
+	private function next_balanced_closer(): bool {
+		$depth    = 0;
+		$tag_name = $this->get_tag();
+
+		if ( ! $this->has_and_visits_its_closer_tag() ) {
+			return false;
+		}
+
+		while ( $this->next_tag(
+			array(
+				'tag_name'    => $tag_name,
+				'tag_closers' => 'visit',
+			)
+		) ) {
+			if ( ! $this->is_tag_closer() ) {
+				++$depth;
+				continue;
+			}
+
+			if ( 0 === $depth ) {
+				return true;
+			}
+
+			--$depth;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks whether the current tag has and visits a closer tag.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @access private
+	 *
+	 * @return bool Whether the current tag has a closer tag.
+	 */
+	public function has_and_visits_its_closer_tag(): bool {
+		$tag_name = $this->get_tag();
+		return ! WP_HTML_Processor::is_void( null !== $tag_name ? $tag_name : '' ) &&
+			! in_array( $tag_name, self::TAGS_THAT_DONT_VISIT_CLOSER_TAG, true );
```

Comment:
Why pass an empty string to `WP_HTML_Processor::is_void()`? It will always return `false` in that case. If so, then it could be something like:

```suggestion
		return null === $tag_name || (
			! WP_HTML_Processor::is_void( $tag_name ) &&
			! in_array( $tag_name, self::TAGS_THAT_DONT_VISIT_CLOSER_TAG, true )
		);
```

But should this method even get called if the processor is at a point where `$tag_name` is `null`? Is it possible for it to be `null` given how it is used here?

Reply:
Improved in https://github.com/WordPress/wordpress-develop/pull/5953/commits/56a4d48e7791b39ef8c73985e043b59fe7d21b0d.

Reply:
I changed the logic so it now returns false if `$tag_name` is `null`.